### PR TITLE
Close #22043: Create widget for progress bar

### LINKED ISSUE #22043: Create widget for progress bar
A widget should be made for the bars seen in the Guest window (used for showing the needs of the guest) and the Ride window (used for showing reliability and down-time).

Besides the usual properties of width and height, this should have the following properties:
- percent filled
- colour
- blinking bounds - if the percentage is between these values, the bar should blink.

### PR DESCRIPTION
**Pull request by Gymnasiast**
Creates a widget for progress bars, as used by the Ride and Guest windows. 

afbeelding
afbeelding

I also have a commit to add them to the Network Status window, but since @AaronVanGeffen is already working on that, I have not included it in this branch. This is what it looks like, though:

afbeelding
afbeelding

### CODE REVIEW

File: src/openrct2-ui/interface/Widget.cpp [hunk 1]
```diff
@@ -1189,6 +1195,41 @@ static void WidgetTextBoxDraw(DrawPixelInfo& dpi, WindowBase& w, WidgetIndex wid
     }
 }
 
+static void WidgetProgressBarDraw(DrawPixelInfo& dpi, WindowBase& w, WidgetIndex widgetIndex)
+{
+    const auto& widget = w.widgets[widgetIndex];
+
+    ScreenCoordsXY topLeft{ w.windowPos + ScreenCoordsXY{ widget.left, widget.top } };
+    //    ScreenCoordsXY bottomRight = topLeft + ScreenCoordsXY{ 149, 8 };
```
Comment (Gymnasiast) [inline src/openrct2-ui/interface/Widget.cpp:1203]: Need to remove the commented-out code.

**Comment by AaronVanGeffen** [comment]
Good to see a special widget type for these kinds of bars. I imagine it'll be useful for plugins as well.

I would prefer the bar not be added to the NetworkInfo window, however. As you know, I have been working on graphical progress windows for denoting loading progress. In my branch, I am introducing a dedicated window for these progress bars, leaving NetworkInfo for the basic network info messages it was once designed to display.

**Comment by Gymnasiast** [comment]
As I already mentioned, these are in a commit of their own that can be very easily reverted. I could leave it out, but I’d rather only do so if your branch is definitely ready for the release.

**Comment by AaronVanGeffen** [comment]
I'd rather we dedicate this PR to just the introduction of the widget type. If my progress bars aren't ready for release, you could file a PR just to introduce these bars in NetworkInfo.

**Comment by Gymnasiast** [comment]
I have taken the commit out and updated the opening post accordingly.

**Comment by AaronVanGeffen** [comment]
Thanks. Have you already made a plugin with which the progress bar widget can be tested, by any chance?

**Comment by Gymnasiast** [comment]
I haven’t, no. That’s maybe something that @Basssiiie could pick up in a follow-up PR, if he likes.

**Comment by AaronVanGeffen** [comment]
Ah, I had assumed you had this working because `ScWidget.hpp` had been changed. I was under the impression the request was for plugin widget functionality in the first place, to be honest. I'm fine with merging this refactor separately first, though. I can take it for a quick test run in a bit.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**